### ISSUE TITLE
Docker Desktop 4.28.0 not starting on Windows 11

### ISSUE THREAD

**Issue by Tsingis** [docker/for-win#13971]
### Description

Docker Desktop 4.28.0 not starting on Windows 11.

### Reproduce

Install Docker Desktop 4.28.0 with installer and/or upgrade from version 4.27.2.

Running on Windows 11 Version 23H2 (OS build 22631).

### Expected behavior

Docker Desktop should start as it did on earlier version 4.27.2

### docker version

```bash
Client:
 Cloud integration: v1.0.35+desktop.11
 Version:           25.0.3
 API version:       1.44
 Go version:        go1.21.6
 Git commit:        4debf41
 Built:             Tue Feb  6 21:13:02 2024
 OS/Arch:           windows/amd64
 Context:           default
```

### docker info

```bash
Client:
 Version:    25.0.3
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.12.1-desktop.4
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.24.6-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  debug: Get a shell into any image or container. (Docker Inc.)
    Version:  0.0.24
    Path:     C:\Program Files\Docker\cli-plugins\docker-debug.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.22
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.4
    Path:     C:\Program Files\Docker\cli-plugins\docker-feedback.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.0.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scout: Docker Scout (Docker Inc.)
    Version:  v1.5.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 0
 Server Version: 25.0.3
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc version: v1.1.12-0-g51d5e94
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
 Kernel Version: 5.15.146.1-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 6
 Total Memory: 7.727GiB
 Name: docker-desktop
 ID: 2ee3cd9d-8e0f-41ab-8b28-daa9ebcc91fa
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

B134AB12-4C51-49E3-975E-D563F2B1A99D/20240313104523

### Additional Info
Lots of processes start up but nothing visible.
![image](https://github.com/docker/for-win/assets/38571453/f90cefae-746e-4110-950c-61d6930a4e46)

Downgrading back to 4.27.2 and no starting issues.

Does not prevent using docker totally fortunately.

**Comment by IssacKramer** [comment]
+1 same for me.
Same issue like this:  https://github.com/docker/for-win/issues/13641
Thought we already over it....

**Comment by TheDoubleH** [comment]
+1 same for me 

and despite uninstalling, I now have the same issue when I try installing 4.27

**Comment by MrBeck-GIT** [comment]
+1 same for me
After restarting Windows, Docker comes up this message. Uninstalling Docker and installing it again fixed it yesterday, until the next Windows reboot.
My specs:
Windows 10 22H2 
Docker 4.28.0

Windows Eventlog Error Entries:
failed wait [error=hcs::Process::waitBackground 6aac43d62bfa7f948b720ac1d9481b6306d5d64ff4e115eca9bdda1c787ca911:21620: hcsshim: the handle generating this notification has been closed]

EDIT:
Uninstalling and reinstalling fixed it again today. After reboot it fails again.

**Comment by TheDoubleH** [comment]
> +1 same for me After restarting Windows, Docker comes up this message. Uninstalling Docker and installing it again fixed it yesterday, until the next Windows reboot. My specs: Windows 10 22H2 Docker 4.28.0
> 
> Windows Eventlog Error Entries: failed wait [error=hcs::Process::waitBackground 6aac43d62bfa7f948b720ac1d9481b6306d5d64ff4e115eca9bdda1c787ca911:21620: hcsshim: the handle generating this notification has been closed]

Exactly the same as my experience - worked for a day, until a reboot

**Comment by sc-ivanlieckens** [comment]
> > +1 same for me After restarting Windows, Docker comes up this message. Uninstalling Docker and installing it again fixed it yesterday, until the next Windows reboot. My specs: Windows 10 22H2 Docker 4.28.0
> > Windows Eventlog Error Entries: failed wait [error=hcs::Process::waitBackground 6aac43d62bfa7f948b720ac1d9481b6306d5d64ff4e115eca9bdda1c787ca911:21620: hcsshim: the handle generating this notification has been closed]
> 
> Exactly the same as my experience - worked for a day, until a reboot

Same issue, Docker Desktop only works as long as no reboot is done, only a full reinstall + reset to factory will return it to functional state until next reboot. I found something else in the event viewer however:

```
Error running exec 97df2d6846baffbb2b6ff1e7b1a1b18cc3558dc0afc56bfe94eaa5b31051e6d8 in container: container d2a4b28b283a8609bdefcec941e34598dcf9e4752701db490d8479e7fe852ad9 encountered an error during hcs::System::CreateProcess: cmd.exe /C "ECHO 192.168.187.178    host.docker.internal >> %systemroot%\system32\drivers\etc\hosts & ECHO 192.168.187.178    gateway.docker.internal >> %systemroot%\system32\drivers\etc\hosts": failure in a Windows system call: The user name or password is incorrect. (0x52e)
[Event Detail:  Provider: 00000000-0000-0000-0000-000000000000]
[Event Detail:  Provider: 00000000-0000-0000-0000-000000000000]
[Event Detail: onecore\vm\compute\management\orchestration\vmhostedcontainer\processmanagement.cpp(173)\vmcomputeagent.exe!00007FF6E4829F4B: (caller: 00007FF6E47DE13A) Exception(3) tid(4c0) 8007052E The user name or password is incorrect.
    CallContext:[\Bridge_ProcessMessage\VmHostedContainer_ExecuteProcess] 
 Provider: 00000000-0000-0000-0000-000000000000]
```

**Comment by dmitryRomanchenko** [comment]
the same issue
Windows 11 Pro 23H2
Docker 4.28.0

**Comment by bsousaa** [comment]
@dmitryRomanchenko @sc-ivanlieckens @TheDoubleH @MrBeck-GIT @IssacKramer can you please share a diagnostics id?

**Comment by MrBeck-GIT** [comment]
diagnostics id:
6A3F91A0-7EB2-4192-B3CF-133A4AFD607E/20240314112945

**Comment by MrBeck-GIT** [comment]
Here is something, that maybe help:

As I already mentioned, if I uninstall Docker and reinstall it, it works again. But my CPU is always at maximum Ghz which results in a noisy fan. Once I send Docker Desktop into the tray, I can not open it anymore, but the containers are running.
The only solution to get CPU fan quiet is to stop "Docker Engine" and "Docker Desktop Service" und kill all tasks related to docker. By doing this, I discovered a solution to the problem:

If you stop both services and run Docker Desktop, then the known message "Docker desktop is not running. would you like to start it? Windows will ask you for elevated access." appears. But if you then click an "Start service" the "Docker Desktop Privileged Helper" asks for permission to start them. After this Docker is running normally, without stressing the CPU. This even works after reboot. Reboot -> Stop Services -> run Docker Desktop -> hit "Start service" -> works
If you stop the services after running and quitting Docker Desktop and they were started by the privileged Helper, you can run Docker Desktop again, without the message and it starts the services by itself.

**Comment by landism** [comment]
@Tsingis Thanks for the report and diagnostics! From the look of things, you've found a bug in Docker Desktop 4.28.0 that causes it to fail on startup when there is an environment variable named `ENVIRONMENT` set. Can you please check if this is set on your machine, and if so, unset it and see if that resolves the problem?

**Comment by Tsingis** [comment]
@landism Yup. Renamed the mentioned env variable and everything works now.

**Comment by IssacKramer** [comment]
My Diagnostics ID:
52013696-9709-4910-8C1A-7B6904B6D2BD/20240314152555

The workaround of @MrBeck-GIT  is WORKING (for now)
Thx @MrBeck-GIT !
But need to see the next days how it goes.

------------------------------------------------------
**Update 6 hours later:**
After boot i still get the annoying popup message and docker desktop again stuck .
I need to do the same trick: stop the 2 services of docker from services panel , launch again docker desktop from the icon, and the elevated window is popup and from that docker started to work again.
Tedious and annoying.

**Comment by TheDoubleH** [comment]
@landism - I don't have my Diagnostic ID.

I kept getting this as well:

![image](https://github.com/docker/for-win/assets/24259184/e8e82ada-1963-4f98-9590-f5629fdfaa3b)

it wasn't until I deleted all of the folders described in the docs. (https://docs.docker.com/desktop/uninstall/) I was able to successfully re-install 4.27.0. 

However when changing to Windows Containers, Docker Desktop just crashed.

I then re-installed again, but unchecking `Use WSL` and now I am back in business.


ps - I have no `ENVIROMENT` enviroment variables

![image](https://github.com/docker/for-win/assets/24259184/cc4b8eb3-7f2b-48fe-9346-9a24258d7684)

**Comment by JanBDL** [comment]
I also have the same problem since 4.28. (I have no "ENVIRONMENT" variable.)
My workaround: kill all \*dock\* processes and then start Docker Desktop again.

**Comment by MrBeck-GIT** [comment]
I also have no "ENVIRONMENT" variable set. The issue started spontaneously. I updated to Docker 4.28 a while ago and it worked without any problem. I changed nothing. The only thing that was installed the day before this started, was Windows Update KB5035845 und Servicing Stack 10.0.19041.4163

**Comment by landism** [comment]
If you have "service is not running", that's #13954. If you don't have an environment variable named "ENVIRONMENT" or the "service is not running" message, can you please open a new ticket and include a [diagnostics id](https://docs.docker.com/desktop/troubleshoot/overview/#diagnose-from-the-app)?
Thanks!

**Comment by remoorejr** [comment]
Here's my workaround: In services, set Docker Engine service to Manual start, Docker Desktop Service should already be set to Manual start. Then when I re-boot and fire up Docker Desktop, I'm prompted to start the Docker Desktop service. Problem solved.

**Comment by krzysztof-nawara-bp** [comment]
Looks like I'm suffering  from the same issue. Not sure if that's relevant, but these are the last lines in vm/init.log:
`[2024-03-16T19:03:46.808584766Z][init][I] [2024-03-16T19:03:46.808511867Z][init][I] Ran init in 160.434872ms
[2024-03-16T19:03:46.824260683Z][init][I] [2024-03-16T19:03:46.824198507Z][init][I] Couldn't start containerd services starting task 01-docker: failed to create task: failed to create shim task: OCI runtime create failed: writing file `devices.allow`: Invalid argument: unknown`

**Comment by ulfk** [comment]
Same problem for me: 

- Using Docker Desktop on Windows 11. 
- Version 4.28.0 does not start,
- Several (dead?) "Docker Desktop.exe" instances in the task manager 
- In an electron logfile of Docker, lots of these errors are piling up: 
  `error   Unhandled uncaught exception : {}`
  - (not very helpfull though)
  - (logfile is C:\Users\UserName\AppData\Local\Docker\log\host\electron-YYYY-MM-DD.log)
- Downgrading to version 4.27.2 resolves the issue.

**Comment by Velociapcior** [comment]
I have the same problem, I'm in the process of downgrading to 4.27.2

**Comment by bsousaa** [comment]
@ulfk @Velociapcior @krzysztof-nawara-bp @remoorejr can you confirm if you have an environment variable named "ENVIRONMENT" in your system? Can you share a diagnostics id?

**Comment by ulfk** [comment]
@bsousaa Yes we do have an "ENVIRONMENT" variable in place. 
Removing/Renaming that variable would require a lot of work and thus is not practical for us.
The diagnostics ID is: F8EDC8CD-49A0-4E93-BEFF-2075F461A759/20240319161537 (created with v4.27.2, currently no problem with that version)
But as our Company Portal (for software we use) is forcing an update to v4.28.0 every day, I currently need to downgrade each morning to v4.27.2 again...

**Comment by krzysztof-nawara-bp** [comment]
@bsousaa I checked and there was no ENVIRONMENT variable. The downgrade to my previously installed version (4.26.1) didn't help by itself, I had to also remove all Docker settings from AppData\ folder
My diagnostic ID is A5FA8540-0B95-4FD0-89B7-B6131ED21112/20240320085120. But I created it right now, after the downgrade (I'm not experiencing any issues).

**Comment by bsousaa** [comment]
@krzysztof-nawara-bp, thank you. Diagnostics logs are persistent so they will include the data from 4.28. You confirm you that Docker Desktop wouldn't open and processes would pile in the task manager?

**Comment by adnansis** [comment]
Are there any news to when is new minor version with bugfixes coming out? Downgrading from 4.28.0 is not an option for me and 4.28.0 is unusable on Windows :(

**Comment by yauhen-vastraknutau-epam** [comment]
> Here's my workaround: In services, set Docker Engine service to Manual start, Docker Desktop Service should already be set to Manual start. Then when I re-boot and fire up Docker Desktop, I'm prompted to start the Docker Desktop service. Problem solved.

Works to me, thanks @remoorejr

**Comment by KumaranFabric** [comment]
Same problem here with starting 4.28.0 on windows 11.  I have noticed that if i kill `dockerd.exe` and restart docker desktop I am able to run normally until the next reboot.

**Comment by bastianschwarz** [comment]
TL;DR: Deleting the ~/.docker/features.json symlink and creating an empty features.json instead solved the issue for me.

Chiming in here: I had this problem for half a year now. Whenever the features.json existed, Docker Desktop refused to start. Intermediate fix: I created an alias in my WSL to delete the file.

Today I took another look and saw that it now creates a Symlink to the Windows user home. There the features.json doesn't exist. So I created it. Same error. I placed an empty object "{}" into it, same error.

It still created the symlink even if the target didn't exist, I noticed the config.json wasn't a symlink. Backend Error log says it cant' open the file ... so I tried creating an empty features.json in my WSL. And now Docker Desktop starts without a hitch.

Not sure what happens with the next update when it tries to migrate and I assume the intent was to reuse the same features.json within multiple WSL distros but ... yeah, here we are.

**Comment by jason-hess** [comment]
I am experiencing this issue as well.  The workaround for me is:

1. Launch Docker Desktop
2. Click `Start Service` when told that `Docker Desktop service is not running`
3. After whatever state Docker Desktop ends up in (sometimes, the GUI launches with an error, sometimes I see no Docker Desktop windows), launch the Task Manager and kill the `dockerd` process, and then any other processes that start with `Docker` including `Docker Desktop`
4. Launch Docker Desktop.  Now when presented with the `Docker Desktop service is not running`, I click `Start service` and the UAC prompt appears and afterward everything is up and running.

**Comment by brasky** [comment]
What a mess, I go on vacation and come back and docker desktop is just dead. Setting docker desktop and docker engine services to manual, then launching docker desktop as admin, and then clicking "Start" when getting the "service is not running" pop up seems to have worked.... 

Reinstalling docker desktop and renaming settings.json (both suggestions I have read in other places like #13954 ) did not work.

**Comment by ghost** [comment]
> Here's my workaround: In services, set Docker Engine service to Manual start, Docker Desktop Service should already be set to Manual start. Then when I re-boot and fire up Docker Desktop, I'm prompted to start the Docker Desktop service. Problem solved.

I have same issues and this workaround worked for me.

**Comment by ulfk** [comment]
@bsousaa and all FYI: we were able to remove the environment variabale "ENVIRONMENT" and now v4.28.0 is starting/running as expected.

**Comment by bsousaa** [comment]
This will be fixed with 4.29 which should release pretty soon

**Comment by Jet0718** [comment]
I had same issue, just install 4.29 replace 4.28, then the issue be solve. now is work normaly.

**Comment by bsousaa** [comment]
Closing the issue now has this has been fixed with 4.29.

**Comment by EvgenyMuryshkin** [comment]
@bsousaa I just upgraded to 4.29 and docker does not run. Downgraded to 4.28 for now.

**Comment by FredoAR** [comment]
same error with docker desktop 4.31.1 and 4.30 in windows 11 Pro V23H2

**Comment by MrBeck-GIT** [comment]
@FredoAR 
Change the Docker services to manual start and let Docker Desktop start the services as you launch it. This issue occurs from
time to time even it should be fixed. After setting the services to manual start I never experienced the issue anymore. Just after a Docker update, which resets the start settings for the services. Then you have to change it back to manual.

**Comment by byronkenoly** [comment]
facing same issue
